insta_reels_publishing_api_sample/index.js

```diff
@@ -77,21 +77,55 @@ app.get("/callback", async function (req, res) {
 // Pages route to retrieve FB OAuth page tokens
 app.get("/pages", async function (req, res) {
     const associatedPagesUri = `https://graph.facebook.com/v14.0/me/accounts?access_token=${req.session.userToken}`;
-    if (req.session.userToken) {
-        try {
-            const associatedPages = await axios.get(associatedPagesUri);
-            req.session.pageData = associatedPages.data.data;
-            res.render('upload_page', {
-                'pages': req.session.pageData
-            });
-        } catch (error) {
-            res.render("index", {
-                error: `There was an error with the request: ${error}`,
-            });
-        }
-    } else {
+    if (!req.session.userToken) {
         res.render("index", { error: "You need to log in first" });
+        return;
     }
+
+    // Request all associated Facebook pages
+    let pagesData = [];
+    try {
+        const accountsResponse = await axios.get(associatedPagesUri);
+        pagesData = accountsResponse.data.data;
+    } catch (error) {
+        res.render("index", {
+            error: `There was an error requesting the FB pages: ${error}`,
+        });
+        return;
+    }
+
+    // Retrieve the Instagram Businesses associated with each page, if any, in a single HTTP request
+    const batchParamValue = pagesData.map(pageData => ({
+        method: "GET",
+        relative_url: `${pageData.id}/?fields=instagram_business_account{name,username}`,
+        access_token: pageData.access_token,
+    }));
+    let batchResponses;
+    try {
+        batchResponses = await axios({
+            method: 'POST',
+            url: `https://graph.facebook.com/?access_token=${req.session.userToken}&include_headers=false`,
```

Comment:
"/?" can be replace by just "?"

Reply:
It could (meaning that it works), but it would not be compliant with [the HTTP URL scheme spec](https://www.rfc-editor.org/rfc/rfc1738#section-3.3) so let's leave this one as is.